test/end-to-end/offer.js

```diff
@@ -295,77 +418,51 @@ describe("Offer endpoint tests", () => {
                     const res = await request()
-                        .get("/offer");
+                        .get("/offers");
 
                     expect(res.status).toBe(HTTPStatus.OK);
                     expect(res.body).toHaveLength(1);
                     // Necessary because jest matchers appear to not be working (expect.any(Number), expect.anthing(), etc)
                     const extracted_data = res.body.map((elem) => {
                         delete elem["_id"]; delete elem["__v"]; delete elem["owner"]; return elem;
                     });
-                    const prepared_test_offer = { ...test_offer };
+                    const prepared_test_offer = {
+                        ...test_offer,
+                        isHidden: false,
+                        // JSON.parse->JSON.stringify needed because comparison below fails otherwise. Spread operator does not work
+                        company: JSON.parse(JSON.stringify(test_company.toObject()))
+                    };
                     delete prepared_test_offer["owner"];
 
                     expect(extracted_data).toContainEqual(prepared_test_offer);
                 });
-            });
-
-            describe("When a `limit` is given", () => {
-                const test_offer = {
-                    title: "Test Offer",
-                    publishDate: "2019-11-22T00:00:00.000Z",
-                    publishEndDate: "2019-11-28T00:00:00.000Z",
-                    description: "For Testing Purposes",
-                    contacts: { email: "geral@niaefeup.pt", phone: "229417766" },
-                    jobType: "SUMMER INTERNSHIP",
-                    fields: ["DEVOPS", "MACHINE LEARNING", "OTHER"],
-                    technologies: ["React", "CSS"],
-                    owner: "aaa712371273",
-                    location: "Testing Street, Test City, 123",
-                };
-
-                const N_OFFERS = 5;
-
-                // TODO: Create a mock owner Company for this test
-                beforeAll(async () => {
-                    await Offer.deleteMany({});
-                    const offers = [];
-                    for (let i = 0; i < N_OFFERS; ++i) {
-                        offers.push(test_offer);
-                    }
-                    await Offer.create(offers);
-                });
-
-                afterAll(async () => {
-                    await Offer.deleteMany({});
-                });
 
-                const RealDateNow = Date.now;
-                const mockCurrentDate = new Date("2019-11-23");
-
-                beforeEach(() => {
-                    Date.now = () => mockCurrentDate.getTime();
-                });
+                test("Only `limit` number of offers are returned", async () => {
 
-                afterEach(() => {
-                    Date.now = RealDateNow;
-                });
+                    // Add 2 more offers
+                    await Offer.create([test_offer, test_offer]);
```

Comment:
This shoul probably be in a hook before this test? It seems weird to have this just hanging here.

Reply:
I belive that was the reason to have a describe and then a test like it was before